### PR TITLE
[TS migration] Move FormFieldGroup to Typescript 📦

### PR DESCRIPTION
**Pull request by brittanylau**
### Notion ticket link

[TS migration](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=e47e78d6b54e4120aa13f06bb61640c5)

### Implementation description

- moved FormFieldGroup.jsx → FormFieldGroup.tsx
  - cleaned up some code & fixed an autofill issue (impacting accessibility) along the way
- moved StudentRoles.js → StudentRole.ts and made it an enum
- refactored the `useFormFields` hook out to its own file for reusability in the family sidebar later

### Steps to test

1. `yarn start`
2. verify the reg form shows up as expected

### What should reviewers focus on?

- ts best practices

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by brittanylau** [comment]
fixing an issue with autocomplete on the form inputs, one sec

File: src/hooks/useFormFields.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import DefaultFieldKey from "../constants/DefaultFieldKey";
+import StudentRole from "../constants/StudentRole";
+import { DefaultField, DynamicField } from "../types";
+
+export type DefaultFormField = DefaultField & { role: StudentRole };
```
Comment (Tayef-Shah) [inline src/hooks/useFormFields.ts:6]: What does the `&` operator do here?

**Review by Tayef-Shah — APPROVED** [review]
LGTM! Nice work :)

File: src/hooks/useFormFields.ts (hunk 1 above)
Comment (brittanylau) [reply src/hooks/useFormFields.ts:6]: good q! this is an [intersection type](https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types), which lets us extend the `DefaultField` type and add an additional `role` property

**Pull request merged**